### PR TITLE
feat(recipes): support tags for better organization

### PR DESCRIPTION
**Pull request by chdsbd**

### CODE REVIEW

File: backend/core/models/recipe.py [hunk 1]
```diff
@@ -60,6 +61,12 @@ class Recipe(CommonInfo, SoftDeleteObject):
         default=None,
         help_text="If a clone, the parent this Recipe was cloned from.",
     )
+    tags = ArrayField(
+        base_field=models.TextField(),
+        null=True,
+        default=list,
+        help_text="If a clone, the parent this Recipe was cloned from.",
```
Comment (sbdchd) [inline backend/core/models/recipe.py:68]: looks like a copy pasta

File: frontend/src/components/RecipeTitle.tsx [hunk 2]
```diff
@@ -85,13 +93,33 @@ class RecipeTitle extends React.Component<
     this.toggleEdit()
   }
 
+  handleNewTagChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.persist()
+    this.setState(() => ({ newTag: e.target.value }))
```
Comment (sbdchd) [inline frontend/src/components/RecipeTitle.tsx:98]:
```suggestion
    this.setState({ newTag: e.target.value })
```

Comment (sbdchd) [reply frontend/src/components/RecipeTitle.tsx:98]: I think that avoids the need for `e.persist`

File: frontend/src/search.ts [hunk 3]
```diff
@@ -108,6 +132,17 @@ export function searchRecipes(params: {
...
+    ? flatMap(params.recipes, x => x.tags || [])
+    : []
+  const normalizedQuery = normalizeQuery(params.query)
+  const options = groupCountSortByName(allTags)
+    .filter(x => {
+      return !normalizedQuery || x[0].includes(normalizedQuery)
+    })
+    .map(x => ({ name: x[0], count: x[1] }))
+    .sort((a, b) => ingredientByNameAlphabetical(a.name, b.name))
+  // console.log({ matchType, options, count: groupCountSortByName(allTags) })
```
Comment (sbdchd) [inline frontend/src/search.ts:145]:
```suggestion
```

File: backend/core/models/recipe.py [hunk 4]
```diff
@@ -60,6 +61,12 @@ class Recipe(CommonInfo, SoftDeleteObject):
         default=None,
         help_text="If a clone, the parent this Recipe was cloned from.",
     )
+    tags = ArrayField(
+        base_field=models.TextField(),
```
Comment (sbdchd) [inline backend/core/models/recipe.py:65]: should we be making sure these are unique?

Comment (chdsbd) [reply backend/core/models/recipe.py:65]: we handle that on the UI

**Review by sbdchd — APPROVED** [review]

**Pull request merged**